hls4ml/model/optimizer/passes/bn_quant.py

```diff
@@ -102,24 +100,26 @@ def transform(self, model, node):
             'n_in' : bn_layer.get_attr('n_in'),
             'n_out' : bn_layer.get_attr('n_in'),
             'n_filt' : bn_layer.get_attr('n_filt'),
-            'epsilon' : bn_layer.get_attr('epsilon'),
             'quantize' : quantize,
-            'beta_quantizer' : bn_layer.get_attr('beta_quantizer'),
-            'gamma_quantizer' : bn_layer.get_attr('gamma_quantizer'),
-            'mean_quantizer' : bn_layer.get_attr('mean_quantizer'),
-            'variance_quantizer' : bn_layer.get_attr('variance_quantizer'),
             'Trace' : bn_layer.get_attr('Trace')
         }
         bnbt_layer = model.make_node('BatchNormalizationQuantizedTanh', 'bnbt_' + bn_layer.name, attrs, bn_layer.inputs)
-        # Replace the old BatchNormalization layer with this one
-        model.replace_node(bn_layer, bnbt_layer)
+        bnbt_layer.set_thresholds(bn_layer.get_weights('scale').data, bn_layer.get_weights('bias').data, node.get_attr('threshold',0.5))
+        # Remove the BatchNormalization layer
+        model.remove_node(bn_layer, rewire=True)
+        # Replace the old Activation layer with this one
+        model.remove_node(node, rewire=True)
+        #model.replace_node(node, bnbt_layer)
+        model.insert_node(bnbt_layer)
```

Comment:
Out of curiosity, is there any reason why `replace_node()` wouldn't work here?

Reply:
Given the changes introduced in #367, this now uses `replace_node`